scriptoocloud/src/main/java/edu/asu/diging/scriptoocloud/core/service/impl/GitRepositoryService.java

```diff
@@ -0,0 +1,110 @@
+package edu.asu.diging.scriptoocloud.core.service.impl;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import edu.asu.diging.scriptoocloud.core.data.GitRepositoryRepository;
+import edu.asu.diging.scriptoocloud.core.exceptions.InvalidGitUrlException;
+import edu.asu.diging.scriptoocloud.core.model.GitRepository;
+import edu.asu.diging.scriptoocloud.core.model.impl.GitRepositoryImpl;
+import edu.asu.diging.scriptoocloud.core.service.DeleteFilesService;
+import edu.asu.diging.scriptoocloud.core.service.GitRepositoryManager;
+import edu.asu.diging.scriptoocloud.core.service.JgitService;
+import edu.asu.diging.scriptoocloud.core.service.UrlFormatterUtility;
+
+/*
+ * Clones remote git repositories and facilitates transactional events related to git repositories in file system
+ * 
+ * @author Jason Ormsby
+ */
+
+@Service
+@Transactional
+@PropertySource("classpath:config.properties")
+public class GitRepositoryService implements GitRepositoryManager{
+
+    @Autowired
+    private DeleteFilesService deleteFilesService;
+    
+    @Autowired
+    private UrlFormatterUtility urlFormatter;
+    
+    @Autowired 
+    private GitRepositoryRepository gitRepositoryJpa;
+    
+    @Autowired
+    private JgitService jGitService;
+
+    @Value("${git.repositories.path}")
+    public String path;
+
+   /*
+    * Handles requests to clone remote git repositories and stores an entity, or updates an entity, 
+    * with information related to the request and repository
+    * 
+    * @param   gitUrl       nonmalformed url of remote git repository 
+    * @param   requester    username in current session that made the request
+    */
+    @Override
+    public void cloneRepository(String gitUrl, String requester) throws InvalidGitUrlException, MalformedURLException{
+        String folderName = urlFormatter.urlToFolderName(gitUrl);
+
+        ZonedDateTime creationDate = ZonedDateTime.now();       
+        
+        GitRepositoryImpl repositoryEntity = gitRepositoryJpa.findByUrl(gitUrl);  
+        
+        if(repositoryEntity != null){
+            repositoryEntity.setUrl(gitUrl);
+            repositoryEntity.setRequester(requester);
+            repositoryEntity.setCreationDate(creationDate);
+            repositoryEntity.setFolderName(folderName);
+            File file = new File(path + repositoryEntity.getFolderName());
```

Comment:
you probably want to make sure path ends with `/` or this will fail

Reply:
I've been looking for a good solution for this and I see that it is possible to use groovy plugin to run scripts in pom file. Does this seem like a good solution or do you have a better way I could deal with validity checks on pom property before it's injected? I'm assuming checking validity after injection would be a bad idea since many classes would need to run independent validity checks to ensure pattern of ending with "/"?

Reply:
I think you're overthinking it. How often is the `path` property used? I would either do the check after object construction `@PostConstruct` or if it is used a lot, I would make it available as a bean and do the check in the method that creates the bean `@Bean` annotation in a config class.